File: client-clojure/src/sirqul_io_t_platform/specs/direction_response.clj
(ns sirqul-io-t-platform.specs.direction-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def direction-response-data
  {
   (ds/opt :action) string?
   (ds/opt :bearing) float?
   (ds/opt :bearingDifference) float?
   (ds/opt :heading) string?
   (ds/opt :distance) float?
   (ds/opt :distanceUnits) string?
   (ds/opt :distanceString) string?
   (ds/opt :startLatitude) float?
   (ds/opt :startLongitude) float?
   (ds/opt :endLatitude) float?
   (ds/opt :endLongitude) float?
   (ds/opt :instruction) string?
   (ds/opt :progressStatus) string?
   })

(def direction-response-spec
  (ds/spec
    {:name ::direction-response
     :spec direction-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/DirectionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import DirectionResponse._

case class DirectionResponse (
  action: Option[String],
bearing: Option[Double],
bearingDifference: Option[Double],
heading: Option[String],
distance: Option[Double],
distanceUnits: Option[String],
distanceString: Option[String],
startLatitude: Option[Double],
startLongitude: Option[Double],
endLatitude: Option[Double],
endLongitude: Option[Double],
instruction: Option[String],
progressStatus: Option[String])

object DirectionResponse {
  import DateTimeCodecs._

  implicit val DirectionResponseCodecJson: CodecJson[DirectionResponse] = CodecJson.derive[DirectionResponse]
  implicit val DirectionResponseDecoder: EntityDecoder[DirectionResponse] = jsonOf[DirectionResponse]
  implicit val DirectionResponseEncoder: EntityEncoder[DirectionResponse] = jsonEncoderOf[DirectionResponse]
}
